src/main/java/com/rainbowrave/RainbowRaveMouseTrailOverlay.java

```diff
@@ -0,0 +1,162 @@
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.rainbowrave;
+
+import java.awt.*;
+import java.util.ArrayList;
+
+import net.runelite.api.Point;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayPriority;
+
+class RainbowRaveMouseTrailOverlay extends Overlay
+{
+    private final RainbowRavePlugin rainbowRavePlugin;
+    private final RainbowRaveMouseTrailPlugin plugin;
+    private final RainbowRaveConfig rainbowRaveConfig;
+
+    public RainbowRaveMouseTrailOverlay(RainbowRavePlugin rainbowRavePlugin, RainbowRaveMouseTrailPlugin plugin, RainbowRaveConfig rainbowRaveConfig)
+    {
+        this.plugin = plugin;
+        this.rainbowRavePlugin = rainbowRavePlugin;
+        this.rainbowRaveConfig = rainbowRaveConfig;
+        setPosition(OverlayPosition.DYNAMIC);
+        setPriority(OverlayPriority.HIGH);
+        setLayer(OverlayLayer.ABOVE_WIDGETS);
+    }
+
+    // Function the get the rainbow color for a particular point based on the size of the trail
+    private Color getRainbowColor(int minHue, int trailSize, int position) {
+        double currentPercent = (double) (position - minHue) / (trailSize - minHue);
+        float hue = (float) (currentPercent * (trailSize - minHue) + minHue);
+        return Color.getHSBColor(hue / 100, 1.0f, 1.0f);
+    }
+
+    // Helper method to handle choosing the correct coloring function
+    private Color getColor(int size, int position) {
+        if(rainbowRaveConfig.whichMouseTrailStyle() == RainbowRaveConfig.MouseTrailStyle.PARTYMODE) {
```

Comment:
This would look better as a switch statement.

Reply:
I realized that only Synced uses alternative coloring, so I've updated the function to be more clear.